test/test_centroids.py

```diff
@@ -62,10 +63,46 @@ def test_centroids_from_mean_verts_scrip(self):
         expected_face_x = uxgrid.face_lon.values
         expected_face_y = uxgrid.face_lat.values
 
-        _populate_centroid_coord(uxgrid, repopulate=True)
+        _populate_face_centroids(uxgrid, repopulate=True)
 
         computed_face_x = (uxgrid.face_lon.values + 180) % 360 - 180
         computed_face_y = uxgrid.face_lat.values
 
         nt.assert_array_almost_equal(expected_face_x, computed_face_x)
         nt.assert_array_almost_equal(expected_face_y, computed_face_y)
+
+    def test_edge_centroids_from_triangle(self):
+        """Test finding the centroid of a triangle."""
+        # Create a triangle
+        test_triangle = np.array([(0, 0, 0), (-1, 1, 0), (-1, -1, 0)])
+
+        # Open the dataset and find the centroids
+        grid = ux.open_grid(test_triangle)
+        _populate_edge_centroids(grid)
+
+        # compute edge_xyz for first edge
+        centroid_x = np.mean(grid.node_x[grid.edge_node_connectivity[0][0:]])
+        centroid_y = np.mean(grid.node_y[grid.edge_node_connectivity[0][0:]])
+        centroid_z = np.mean(grid.node_z[grid.edge_node_connectivity[0][0:]])
+
+        # Test the values of computed first edge centroid and the populated one
+        self.assertEqual(centroid_x, grid.edge_x[0])
+        self.assertEqual(centroid_y, grid.edge_y[0])
+        self.assertEqual(centroid_z, grid.edge_z[0])
+
+    def test_edge_centroids_from_scrip(self):
```

Comment:
Can you write a test using one of the MPAS datasets (when loaded, they already have the spherical and Cartesian edge coordinates).

You can call `_construct_edge_centroids` after loading the MPAS mesh and then compare the expected to the computed results.

Reply:
This test was actually labeled wrong and is using an MPAS file, as the SCRIP file doesn't have any edges on it. I will change the name to reflect this, thanks for noticing this!